src/mito2/src/manifest/checkpointer.rs

```diff
@@ -118,6 +120,40 @@ impl Checkpointer {
         self.inner.last_checkpoint_version.load(Ordering::Relaxed)
     }
 
+    /// Update the `removed_files` field in the manifest by the options in `manifest_options`.
+    /// This should be called before maybe do checkpoint to update the manifest.
+    pub(crate) fn update_manifest_removed_files(
```

Comment:
We haven't called this method?

Reply:
> We haven't called this method?

it's called just before `maybe_do_checkpoint`